pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -47,18 +48,16 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 
 // ValidateHost checks that a route's host name satisfies DNS requirements, with
 // the assumption that the caller has already checked for an empty host name.
-// Unless the allowNonCompliant annotation is set to true, host name must have
-// at least two labels, with each label no more than 63 characters from the set of
-// alphanumeric characters, '-' or '.', and must start and end with an alphanumeric
-// character. A trailing dot is allowed. The total host name length must be no more
-// than 253 characters.
-// If allowNonCompliant is set to true, it uses a smaller set of conditions from
-// IsDNS1123Subdomain, e.g. character set as described above, and total host name
-// length must be no more than 253 characters.
-func ValidateHost(host string, allowNonCompliant string, hostPath *field.Path) field.ErrorList {
-	result := field.ErrorList{}
-
-	if allowNonCompliant == "true" {
+// Unless allowNonCompliant is true, host name must have at least two labels,
+// with each label no more than 63 characters from the set of alphanumeric
+// characters, '-' or '.', and must start and end with an alphanumeric
+// character. A trailing dot is allowed.  The total host name length must be no
+// more than 253 characters.  If allowNonCompliant is true, it uses a smaller
+// set of conditions from IsDNS1123Subdomain, e.g. character set as described
+// above, and total host name length must be no more than 253 characters.
+func ValidateHost(host string, allowNonCompliant bool, hostPath *field.Path) field.ErrorList {
```

Comment:
I think the annotation that was formerly used for this parameter is still set to a string, just from:
https://github.com/openshift/openshift-apiserver/blob/master/pkg/route/apis/route/validation/validation.go#L64

I couldn't find any references in documentation though.

Reply:
Semantically, `ValidateHost` really takes a Boolean value.  The fact that the caller may be deriving the value from an annotation is something that the caller can deal with, by converting the string annotation value into a Boolean value before calling `ValidateHost`.  In some cases, there is no annotation, and the caller just needs to be able to pass in a Boolean value.  For these reasons, I think that making the parameter a Boolean and requiring the caller to convert a string value (if that's what it has) into Boolean value for `ValidateHost` makes this function and its semantics cleaner.  

This isn't really related to `ValidateSubdomain`, though, so I can remove the change if you prefer.